sdks/wasm/src/driver.c

```diff
@@ -674,3 +680,142 @@ mono_wasm_enable_on_demand_gc (void)
 {
 	mono_wasm_enable_gc = 1;
 }
+
+MonoAssembly* get_zone_Info_assembly()
+{
+	static MonoAssembly* zoneInfoAssembly;
+
+	if (!try_load_zone_info) {
+
+		fprintf (stdout, "%s\n", "Locating ZoneInfo assembly.");
+		// try to open the ZoneInfo assembly
+		zoneInfoAssembly = mono_domain_assembly_open (root_domain, "WebAssembly.ZoneInfo");
+		// if not open then try to load it.
+		if (!zoneInfoAssembly) {
+			MonoImageOpenStatus status;
+			MonoAssemblyName* aname = mono_assembly_name_new ("WebAssembly.ZoneInfo");
+			zoneInfoAssembly = mono_assembly_load (aname, NULL, &status);
```

Comment:
On-demand initialization requires memory barriers, both in the read and write paths, to account for multiple threads.